File: sdks/sdkwork-craw-chat-sdk-admin/sdkwork-craw-chat-sdk-admin-flutter/composed/lib/src/nodes_module.dart
import 'package:craw_chat_admin_backend_sdk/craw_chat_admin_backend_sdk.dart';

import 'context.dart';

class CrawChatAdminNodesModule {
  final CrawChatAdminSdkContext context;

  CrawChatAdminNodesModule(this.context);

  Future<JsonObject> activate(Identifier nodeId) {
    return context.backendClient.nodes.activateNode(nodeId);
  }

  Future<JsonObject> drain(Identifier nodeId) {
    return context.backendClient.nodes.drainNode(nodeId);
  }

  Future<JsonObject> migrateRoutes(Identifier nodeId, JsonObject body) {
    return context.backendClient.nodes.migrateNodeRoutes(nodeId, body);
  }
}

File: sdks/sdkwork-craw-chat-sdk-admin/sdkwork-craw-chat-sdk-admin-typescript/composed/src/nodes-module.ts
import type { Identifier, JsonObject } from './types.js';
import type { CrawChatAdminSdkContext } from './sdk-context.js';

export class CrawChatAdminNodesModule {
  constructor(private readonly context: CrawChatAdminSdkContext) {}

  activate(nodeId: Identifier): Promise<JsonObject> {
    return this.context.backendClient.nodes.activateNode(nodeId);
  }

  drain(nodeId: Identifier): Promise<JsonObject> {
    return this.context.backendClient.nodes.drainNode(nodeId);
  }

  migrateRoutes(nodeId: Identifier, body: JsonObject): Promise<JsonObject> {
    return this.context.backendClient.nodes.migrateNodeRoutes(nodeId, body);
  }
}
